### PR TITLE
LMDB corruption callback

### PR DESCRIPTION
**Pull request by vpodzime**
Merge together:
https://github.com/cfengine/core/pull/3820
https://github.com/cfengine/buildscripts/pull/614

### CODE REVIEW

**Comment by cf-bottom** [comment]
Thanks for submitting a pull request! Maybe @olehermanse can review this?

**Comment by vpodzime** [comment]
@cf-bottom jenkins with exotics, please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/2717/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/2717/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-2717/

File: libpromises/dbm_lmdb.c [hunk 1]
```diff
@@ -223,6 +223,13 @@ static int LmdbEnvOpen(
     return EBUSY;
 }
 
+static void HandleLDMBCorruption(MDB_env *env, const char *msg)
```
Comment (olehermanse) [inline libpromises/dbm_lmdb.c]: `mittigate` -> `mitigate` in commit message.

**Comment by olehermanse** [comment]
> Alright, I triggered a build:

> [Build Status](https://ci.cfengine.com//job/pr-pipeline/2717/)

> (with exotics)

@vpodzime fr-tests red, did you already fix this?

**Comment by vpodzime** [comment]
> @vpodzime fr-tests red, did you already fix this?

Yeah, that was some weird unrelated failure with PostgreSQL errors in the first *promises.cf* run that's part of the bootstrap. Didn't happen again in two more runs of the FR tests I triggered.

**Pull request merged**